File: rb-appscript/trunk/test/test_findapp.rb
#!/usr/bin/ruby -w

begin; require 'rubygems'; rescue LoadError; end

require 'test/unit'
require "_aem/findapp"


class TC_FindApp < Test::Unit::TestCase

	def test_find
		[
			['/Applications/iCal.app', '/Applications/iCal.app'],
			['ical.app', '/Applications/iCal.app'],
			['ICAL', '/Applications/iCal.app'],
		].each do |val, res|
			assert_equal(res, FindApp.by_name(val))
		end
		assert_equal('/Applications/TextEdit.app', FindApp.by_creator('ttxt'))
		assert_equal('/System/Library/CoreServices/Finder.app', FindApp.by_id('com.apple.finder'))
		assert_raises(FindApp::ApplicationNotFoundError) { FindApp.by_name('NON-EXISTENT-APP') }

		# assert_equal("/Users/has/\306\222\303\270u\314\210.app", FindApp.by_name("\306\222\303\270u\314\210.app")) # utf8 paths work ok
	end
end
		

File: py-appscript/trunk/appscript_3x/test/test_findapp.py
#!/usr/bin/env python3

import unittest
from aem import findapp


class TC_FindApp(unittest.TestCase):

	def test_find(self):
		for val, res in [
			['/Applications/iCal.app', '/Applications/iCal.app'],
			['ical.app', '/Applications/iCal.app'],
			['ICAL.APP', '/Applications/iCal.app'],
			['ICAL', '/Applications/iCal.app'],
		]:
			self.assertEqual(res, findapp.byname(val))
		self.assertEqual('/Applications/TextEdit.app', findapp.bycreator(b'ttxt'))
		self.assertEqual('/System/Library/CoreServices/Finder.app', findapp.byid('com.apple.finder'))
		self.assertRaises(findapp.ApplicationNotFoundError, findapp.byname, 'NON-EXISTENT-APP')

		#self.assertEqual(unicode("/Users/has/\306\222\303\270u\314\210.app", 'utf8'), findapp.byname(unicode("\306\222\303\270u\314\210.app", 'utf8'))) # Unicode paths work ok

if __name__ == '__main__':
	unittest.main()
